ui/job-view/redux/stores/selectedJob.js

```diff
@@ -135,13 +135,22 @@ const doSetSelectedJobFromQueryString = (notify, jobMap) => {
 
   // Try to find the Task by taskId and runID
   if (taskId) {
-    const retryId = parseInt(runId, 10);
-    const task = Object.values(jobMap).find(
-      (job) => job.task_id === taskId && job.retry_id === retryId,
-    );
+    let task;
+    if (runId) {
+      const retryId = parseInt(runId, 10);
+      task = Object.values(jobMap).find(
+        (job) => job.task_id === taskId && job.retry_id === retryId,
+      );
+    } else {
+      const runs = Object.values(jobMap)
+        .filter((job) => job.task_id === taskId)
+        .sort((left, right) => left.retry_id - right.retry_id);
+      task = runs[runs.length - 1];
```

Comment:
If we don't find anything here, we should probably generate an error.

Reply:
To your point, I think this would be the right place to call:

    notify('Error parsing the task id blah blah', 'danger', { sticky: true }

Reply:
We would also want to do a `return` here to prevent checking the database, since we know the `task` string is bad.